src/lucky/routable.cr

```diff
@@ -145,14 +145,51 @@ module Lucky::Routable
     # no-op by default
   end
 
+  NORMALIZED_ROUTES = [] of Nil # types are not resolved in macros
+
+  # :nodoc:
+  macro enforce_route_uniqueness(method, original_path)
+    # Regex for capturing the param part for normalization
+    #
+    # So "/users/:user_id" is changed to "/users/:normalized"
+    {% normalized_path = original_path.gsub(/(\:\w*)/, ":normalized") %}
+
+    {% if already_used_route = NORMALIZED_ROUTES.find { |route| route[:normalized_path] == normalized_path && route[:method] == method } %}
```

Comment:
Not sure if this matter or not, but would it make sense (from a lookup perspective) to make `NORMALIZED_ROUTES` a `Hash` instead? The normalized path would be the key in this case.

Reply:
@kevinsjoberg I tried, but there are some issues with Hash/NameTuples in macro constants that I couldn't seem to get around :(

Also since it needs to match both method *and* normalized path I think this may actually work better. Since if we used the normalized path as key that would mean there is no way to lookup based on method. I suppose we could combine the method and path into one string key though 🤔 Maybe someone can do a cleanup pass later and can figure out how to use the Hash in a macro